proposed/MPID Messaging.md

```diff
@@ -0,0 +1,295 @@
+1. `MpidManagers(A)` send a GetResponse to `MPidClient(B)`.
+1. On receiving the message, `MpidClient(B)` sends a remove request to `MpidManagers(B)` via a Delete.
+1. `MpidManagers(B)` remove the corresponding entry from the inbox for B and forward the remove request to `MpidManagers(A)`.  `MpidManagers(A)` then remove the corresponding entry from the outbox for A.
+
+`MpidClient(A)` might decide to remove the `MpidMessage` from the outbox if the message hasn't been retrieved by `MpidClient(B)` yet.  In this case, `MpidManagers(A)` should not only remove the corresponding `MpidMessage` from their outbox for A, but also send a notification to the group of `MpidManagers(B)` so they can remove the corresponding entry from their inbox of B.  These messages will all be Delete requests.
+
+_MPidClient(A)_ =>> |__MPidManagers(A)__ (Put)(Header.So) *->> | __MPidManagers(B)__  (Store(Header))(Online(MpidClient(B)) ? Header.So : (WaitForOnlineB)(header.So)) *-> | _MpidClient(B)_ So.Retrieve ->> | __MpidManagers(A)__ So.Message *-> | _MpidClient(B)_ Remove.So ->> | __MpidManagers(B)__ {Remove(Header), Remove.So} *->> | __MpidManagers(A)__ Remove
+
+## MPID Client
+
+The MPID Client shall provide the following key functionalities :
+
+1. Send Message (Put from sender)
+1. Retrieve Full Message (Get from receiver)
+1. Remove sent Message (Delete from sender)
+1. Accept Message header (if the "push" model is used) and/or Retrieve Message Header (if the "pull" model is used)
+
+If the "push" model is used, an MPID Client is expected to have its own routing object (not shared with the MAID Client).  In this way it can directly connect to its own MpidManagers, allowing them to know its online status and hence they can push message headers to it as and when they arrive.
+
+Such a separate routing object is not required if the "pull" model is used.  This is where the MPID Client periodically polls its network inbox for new headers.  It may also have the benefit of saving the battery life on mobile devices, as the client app doesn't need to keep MPID Client running all the time.
+
+## Planned Work
+
+1. Vault
+    1. outbox
+    1. inbox
+    1. sending message flow
+    1. retrieving message flow
+    1. deleting message flow
+    1. churn handling and refreshing for account_transfer
+    1. detecting of Client, if "push" model used
+    1. MPID Client addressing (if MPID address registration procedure is to be undertaken - i.e. in "pull" model)
+
+1. Routing
+    1. `Authority::MpidManager`
+    1. definition of `MPID_MESSAGE_TAG` and `MPID_HEADER_TAG`
+    1. definition of `MpidMessage` and `MpidHeader`
+    1. support Delete (for StructuredData only)
+    1. if "push" model: notify vault when a client joins and notifying Client with `MpidHeader`
+    1. address relocation (allows Client to use a fixed MPID address to connect to the network and allows "push" model - not required for "pull" model)
```

Comment:
Well, the pull model was our preference anyway, so I guess we can update the RFC to target the pull model, and maybe just mention the push model as future work?